Introduce api-gateway chart (#5425)
kyma-project/kyma: resources/api-gateway/README.md

# API-Gateway

## Overview

> API-Gateway is a component allowing exposure of a service through the kyma Console. It deploys and manages Istio and Ory/Oathkeeper CRDs.

Reviewer: ```suggestion
API-Gateway is a component that allows exposing services through the kyma Console. It deploys and manages Istio and Ory/Oathkeeper Custom Resource Definitions (CRDs).
```

> This chart installs a controller, which requires and uses the following applications/CRD, which should be installed beforehand:

Reviewer: ```suggestion
This chart installs the controller, which requires these CRDs to expose services:
```
a -> the - this chart installs a specific controller, not just any random controller ;)
removed "applications" - I see only CRDs listed here

- Istio [VirtualService](https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/)
- Istio [Policy](https://istio.io/docs/reference/config/istio.authentication.v1alpha1/)
- Oathkeeper [AccessRule](https://www.ory.sh/docs/oathkeeper/)

>   - Oathkeeper CRD resources are available as charts in [this repo](https://github.com/ory/k8s)

Reviewer: Make this a note. 

`>**NOTE:** Oathkeeper CRD resources are available as charts in [this](https://github.com/ory/k8s) repository.`
